NossoMercadoLivre/Controllers/UserController.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using NossoMercadoLivre.Models.ViewModels;
+using NossoMercadoLivre.Services;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace NossoMercadoLivre.Controllers
+{
+    [Route("api/User")]
+    [Produces("application/json")]
+    public class UserController : Controller
+    {
+        private readonly IUserService _userService;
+
+        public UserController(IUserService userService)
+        {
+            _userService = userService;
+        }
+
+        [HttpPost("Create")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(typeof(ErrorResponseViewModel), 400)]
+        public IActionResult Create([FromBody] CreateUserViewModel model)
```

Comment:
*Separamos as bordas externas do sistema do seu núcleo. Não ligamos parâmetros de requisição externa com objetos de domínio diretamente, assim como não serializamos objetos de domínio para respostas de API.*

Essa é uma prática legal e já estabelecida no asp.net mvc/core, acho bem boa.

Reply:
Na real, lá no meu trabalho ta cheio de código que liga o domínio direto, apesar de muitos utilizarem viewModel também, mas o conceito mesmo, eu aprendi lendo seu manifesto um tempo atrás.

Reply:
Vixe, legal que você gostou então. Se for olhar pelo lado da carga cognitiva, até aumenta um pouco. Por outro lado o histórico mostra que ligar domínio direto aqui tende a gerar um acoplamento muito chato e deixar o domínio bem mais frágil.